src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -143,16 +142,31 @@ void AddImports(IReadOnlyList<IProjectImportSnapshot> value)
             AddImports(e.Value);
         }
 
-        private void OnSourceItemChanged(IProjectSubscriptionUpdate e)
+        private void OnSourceItemChanged(IProjectSubscriptionUpdate e, IProjectItemSchema projectItemSchema)
         {
-            foreach (var itemType in e.ProjectChanges.Where(changes => changes.Value.Difference.AnyChanges))
+            var itemTypes = new HashSet<string>(projectItemSchema.GetKnownItemTypes().Where(itemType => projectItemSchema.GetItemType(itemType).UpToDateCheckInput));
```

Comment:
Seems like we wont be hitting this path often, so this does not seem to be a perf hit.But would we be performant in the case of loading a project with a huge number of files?